Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -20,10 +20,69 @@ private static void Test(string args, int expected)
             Console.WriteLine($"{args} : {result}");
         }
 
+
         public static int FindDigit(string equation)
-        {
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            int result = -1;
+            string temp;
+            int operandone, operandtwo, operandthree;
+            string[] operands = equation.Split(new Char[] { '*', '=' });
+            if (operands[2].Contains("?"))
+            {
+                operandone = Int32.Parse(operands[0]);
+                operandtwo = Int32.Parse(operands[1]);
+                operandthree = operandone * operandtwo;
+                temp = operandthree.ToString();
+                if (temp.Length != operands[2].Length)
+                    return -1;
+                else
+                {
+                    result = FindMissingDigit(temp, operands[2]);
+
+                }
+            }
+            else if (operands[0].Contains("?"))
+            {
+                result = GetLeftSideOperandDigit(operands[0], operands[1], operands[2]);

```

Comment:
The logic to find the missing digit is similar for all the operands. Try if you can use one method.